File: apps/api/routes/workspaces/invitations/list_invitations.py
# apps/api/routes/workspaces/invitations/list_invitations.py

"""Route for listing workspace invitations."""

from typing import Annotated
from uuid import UUID

from fastapi import APIRouter, Path, Query

from core.dependencies import AsyncDbSessionDep, CurrentUserDep
from services.workspaces.invitations import list_invitations as list_invitations_service
from services.workspaces.schemas import WorkspaceInvitationsListResponse

router = APIRouter()


@router.get("/{workspace_id}/invitations")
async def list_invitations(
    db: AsyncDbSessionDep,
    actor: CurrentUserDep,
    workspace_id: Annotated[UUID, Path()],
    include_accepted: Annotated[bool, Query()] = False,
    include_expired: Annotated[bool, Query()] = False,
    limit: Annotated[int, Query(ge=1, le=500)] = 100,
    offset: Annotated[int, Query(ge=0)] = 0,
) -> WorkspaceInvitationsListResponse:
    return await list_invitations_service(
        db,
        actor=actor,
        workspace_id=workspace_id,
        include_accepted=include_accepted,
        include_expired=include_expired,
        limit=limit,
        offset=offset,
    )

File: apps/web/src/features/workspaces/api/list-invitations.ts
// apps/web/src/features/workspaces/api/list-invitations.ts

import { queryOptions, useSuspenseQuery } from "@tanstack/react-query"

import { apiRequest } from "@/lib/api/client"
import type { WorkspaceInvitationsListResponse } from "@/features/workspaces/types"

export function workspaceInvitationsQueryKey(workspaceId: string) {
  return ["workspaces", workspaceId, "invitations"] as const
}

async function listInvitations(workspaceId: string) {
  return apiRequest<WorkspaceInvitationsListResponse>(`/workspaces/${workspaceId}/invitations`, {
    query: {
      include_accepted: false,
      include_expired: false,
      limit: 100,
      offset: 0,
    },
  })
}

function workspaceInvitationsQueryOptions(workspaceId: string) {
  return queryOptions({
    queryKey: workspaceInvitationsQueryKey(workspaceId),
    queryFn: () => listInvitations(workspaceId),
    staleTime: 30_000,
  })
}

export function useWorkspaceInvitationsQuery(workspaceId: string) {
  return useSuspenseQuery(workspaceInvitationsQueryOptions(workspaceId))
}
